subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java

```diff
@@ -280,6 +281,10 @@ private void populateFromComponent() {
             Set<MavenDependencyInternal> dependencies = dependenciesFor(usageContext);
             for (ModuleDependency dependency : usageContext.getDependencies()) {
                 if (seenDependencies.add(PublishedDependency.of(dependency))) {
+                    if (dependency.getArtifacts().isEmpty() && getCoordinates().getModule().equals(DefaultModuleIdentifier.newId(dependency.getGroup(), dependency.getName()))) {
```

Comment:
:o: Some test coverage to pin down this behavior would be good.

Reply:
I've added some more tests 😉 
And fixed an issue at the same time, so thanks for pushing me!

Reply:
:+1: PR looks good to merge then.